ab2d-aggregator/src/main/java/gov/cms/ab2d/aggregator/FileUtils.java

```diff
@@ -0,0 +1,186 @@
+package gov.cms.ab2d.aggregator;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.channels.FileChannel;
+import java.nio.file.Files;
+import java.nio.file.NoSuchFileException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static gov.cms.ab2d.aggregator.FileOutputType.getFileType;
+import static java.nio.file.StandardOpenOption.CREATE;
+import static java.nio.file.StandardOpenOption.READ;
+import static java.nio.file.StandardOpenOption.WRITE;
+
+/**
+ * We work a lot with files so having a utils class to help with that is useful
+ */
+@SuppressWarnings("PMD.DataflowAnomalyAnalysis")
+public class FileUtils {
+    /**
+     * Given a list of files, combine them into an outfile and then remove the component files
+     *
+     * @param files - the files to combine
+     * @param outFileName - the output file name with location
+     * @throws IOException - if we have any IO funny business
+     */
+    public static void combineFiles(List<File> files, String outFileName) throws IOException {
+        Path outFile = Paths.get(outFileName);
+        try (FileChannel out = FileChannel.open(outFile, CREATE, WRITE)) {
+            for (File file : files) {
+                Path inFile = Paths.get(file.getAbsolutePath());
+                try (FileChannel in = FileChannel.open(inFile, READ)) {
+                    for (long p = 0, l = in.size(); p < l;)
+                        p += in.transferTo(p, l - p, out);
+                }
+            }
+        }
+    }
+
+    /**
+     * Delete a list of files
+     *
+     * @param files - The files to delete
+     * @return true if all the files were deleted
+     */
+    public static boolean cleanUpFiles(List<File> files) {
+        boolean clean = true;
+        if (files == null || files.isEmpty()) {
+            return true;
+        }
+        for (File f : files) {
+            clean = clean && f.delete();
+        }
+        return clean;
+    }
+
+    /**
+     * Create a directory and any intermediary directories if needed
+     *
+     * @param dir - the directory to create
+     * @return the created directory
+     * @throws IOException if we can't create it
+     */
+    public static File createADir(String dir) throws IOException {
+        File file = new File(dir);
+        if (!file.exists()) {
+            return Files.createDirectories(Path.of(file.getAbsolutePath())).toFile();
+        }
+        return file;
+    }
+
+    /**
+     * Retrieve the data or error files in a driectory and get the total size of all of those files
+     *
+     * @param dirName - the directory to search
+     * @param error - if we are looking for error files or data files
+     * @return the total size of the found files
+     */
+    public static long getSizeOfFiles(String dirName, boolean error) {
```

Comment:
Passing in the desired FileOutputType instead of a boolean simplifies this code a bit.

Reply:
Except that the difference between _error.ndjson and .ndjson can cause issues because technically _error.ndjson would match .ndjson so I put the logic down lower in the code to prevent the issue from arising.